Octokit/Http/HttpClientAdapter.cs

```diff
@@ -198,6 +198,28 @@ public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, Can
             var response = await _http.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
             // Need to determine time on client computer as soon as possible.
             var receivedTime = DateTimeOffset.Now;
+
+            if (response.StatusCode == HttpStatusCode.Forbidden
+                && response.Headers.Contains(ResponseHeaders.RateLimitRemaining)
+                && response.Headers.GetValues(ResponseHeaders.RateLimitRemaining).FirstOrDefault() == "0")
+            {
+                var nowInEpochSecondsUtc = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+                var resetTimeInEpochSecondsUtc = long.Parse(response.Headers.GetValues(ResponseHeaders.RateLimitReset).First());
+
+                var timeToWait = resetTimeInEpochSecondsUtc > nowInEpochSecondsUtc 
+                    ? TimeSpan.FromSeconds(resetTimeInEpochSecondsUtc - nowInEpochSecondsUtc)
+                    : TimeSpan.Zero;
+
+                // Could rate limit refreshes be up to an hour in the future?
+                // Should we conditionally retry only if it's a short period in the future
+                if (timeToWait < TimeSpan.FromSeconds(30))
```

Comment:
This is _quite a long time_ to wait in the case of microservices. Places where I use Octokit would have a request timeout in the region of a few seconds _at most_, so this would be redundant in those cases as the `CancellationToken` would always fire and prevent another attempt ever happening while masking the HTTP request exception from the 403(429?) with a timeout-related exception instead.

Reply:
Yeah I do agree. From the documentation, unless I'm wrong, it sounded like rate limits reset hourly. In which case this functionality might not be particularly useful as it would very rarely be hit.

Reply:
Yeah from my experience, once you hit a limit then you just have to wait, unless it's a secondary rate-limit - those reset much sooner (I don't know how often though, I think it's the order of minutes).